### ISSUE TITLE
Endpoint closure that adds parameters not working for .upload tasks

### ISSUE THREAD

**Issue by ivanmkc** [Moya/Moya#1066]
Moya v8.0.3

I have a custom endpointClosure to add authentication paramters:
```swift
private let endpointClosure = { (target: MyTarget) -> Endpoint<MyTarget> in
        let defaultEndpoint = RxMoyaProvider.defaultEndpointMapping(for: target)
        
        // Sign all non-authenticating requests
        switch target {
            case .LoginByEmail:
                return defaultEndpoint
            default:
                guard let currentSession = try? MyAuthenticationService.sharedInstance.session.value(), let id = currentSession?.id, let token = currentSession?.token else
                {
                    return defaultEndpoint
                }
                
                return defaultEndpoint.adding(newParameters: [
                    "u" : id,
                    "t" : token])
        }
    }
```

This works fine for all my existing requests. I've just added a new target for uploading images, which is an upload task as defined here:

```swift
var task : Task {
        switch self {
            case .UploadImage(let image):
                guard let image = UIImageJPEGRepresentation(image, 1.0) else { return .request }
                
                let formData: [MultipartFormData] = [MultipartFormData(provider: .data(image), name: "pic", fileName: "pic", mimeType: "image/jpeg")]
                
                return .upload(UploadType.multipart(formData))
            default:
                return .request
        }
    }
```

This doesn't work. When I sniff the packets, I noticed that the authentication parameters from endpointClosure are missing. Adding them manually as below works, but this seems very hacky as that's what the endpointClosure is supposed to do for me:

```swift
var parameters: [String: Any]? {
switch self
{
case .UploadImage:
                    guard let currentSession = try? MyAuthenticationService.sharedInstance.session.value(), let id = currentSession?.id, let token = currentSession?.token else
                    {
                        return [:]
                    }
                    
                    return [
                    "u" : id,
                    "t" : token,
                    "type_no" : 1]
}
```

In conclusion, the endpointClosure modifies my parameters successfully for .request tasks but not for .upload tasks

**Comment by sunshinejr** [comment]
Hey @ivanmkc. I've just tested a similar case, using `RxMoyaProvider`, upload task and `endpointClosure` without any conditional statements, and seems like every time Moya is adding correct parameters. By the code there is also no difference whether you are using `upload` or `request` task, our `request` method will just get the endpoint from `endpointClosure` if there is any. 

I'm gonna close this issue for now, however, if you are certain that this issue exist, please create an example project that I can test against, and reopen the issue.
